src/test/java/com/pngencoder/PngEncoderInputTypesTest.java

```diff
@@ -0,0 +1,143 @@
+package com.pngencoder;
+
+import org.junit.jupiter.api.Test;
+
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import javax.imageio.ImageIO;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * LICENCE:
+ * thermos_36667_sm.gif was downloaded from
+ * https://etc.usf.edu/clipart/36600/36667/thermos_36667_sm.gif
+ * It is used as an example of an indexed input with no color depth.
+ * The png encoder may (should?) convert it into a single grayscale channel.
+ *
+ * https://etc.usf.edu/clipart/info/license
+ */
+public class PngEncoderInputTypesTest {
+
+    @Test
+    void testWorkaround() throws IOException {
+        BufferedImage bufferedImage = PngEncoderTestUtil.readTestImageResource("thermos_36667_sm.gif");
+
+        if (bufferedImage.getType() == BufferedImage.TYPE_BYTE_INDEXED) {
```

Comment:
Shouldn't this be always true here?

Reply:
This test can just be removed. I wanted to show how to workaround the issue in client code.